crates/oxc_transformer/src/react/comments.rs

```diff
@@ -0,0 +1,621 @@
+use oxc_ast::{Comment, CommentKind};
+use oxc_syntax::identifier::is_irregular_whitespace;
+
+use crate::{ReactJsxRuntime, ReactOptions, TransformCtx};
+
+/// Scan through all comments and find the following pragmas:
+///
+/// * @jsxRuntime classic / automatic
+/// * @jsxImportSource custom-jsx-library
+/// * @jsxFrag Preact.Fragment
+/// * @jsx Preact.h
+///
+/// The comment does not need to be a JSDoc comment,
+/// otherwise `JSDoc` could be used instead.
+///
+/// This behavior is aligned with Babel.
+pub(crate) fn update_options_with_comments(options: &mut ReactOptions, ctx: &TransformCtx) {
+    for comment in ctx.trivias.comments() {
```

Comment:
With the introduction of leading comments, I think iterating through the first set of leading comments should suffice. There are no specifications but I think we can iterate through less comments.

Reply:
Can experiment in another PR.

Reply:
Ah OK great!